File: Programmers/LEVEL_1/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<bool> eratos(n + 1, false);
    for (int i = 2; i <= n; i++)
    {
        if (eratos[i] == false)
            answer++;
        for (int j = i; j <= n; j += i)
            eratos[j] = true;
    }
    return answer;
}

File: Programmers/LEVEL_1/소수찾기.js
function solution(n) {
  var answer = 0;

  let arr = [];
  arr.push(false);
  arr.push(false);
  for (let i = 2; i <= n; i++) {
    arr.push(true);
  }

  for (let i = 2; i * i <= n; i++) {
    if (arr[i]) {
      for (let j = i * i; j <= n; j += i) {
        arr[j] = false;
      }
    }
  }

  for (let flag of arr) {
    if (flag) answer++;
  }

  return answer;
}
